src/main/java/thePackmaster/cards/magnetizepack/Attraction.java

```diff
@@ -0,0 +1,44 @@
+package thePackmaster.cards.magnetizepack;
+
+import basemod.BaseMod;
+import basemod.helpers.CardModifierManager;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import thePackmaster.actions.magnetizepack.MagnetizeAction;
+import thePackmaster.cardmodifiers.magnetizepack.MagnetizedModifier;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+public class Attraction extends AbstractMagnetizeCard {
```

Comment:
Actually, the draw action has a callback which you can use to apply your modifier, no need for an extra action:
You can look at Scrape from BaseGame or Rummage from PM for an example.